router/core/graphql_handler.go

```diff
@@ -190,6 +190,9 @@ func (h *GraphQLHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 		if errs := resolveCtx.SubgraphErrors(); errs != nil {
 			w.Header().Set("Cache-Control", "no-store, no-cache, must-revalidate")
+			// This is recorded in the defer above, but this is used to make sure the error is accessible to
+			// any response writer wrappers set in custom modules that need it when WriteTo is called
+			reqCtx.SetError(errs)
```

Comment:
I see other paths e.g. `WriteError` where the error would not be set. I think we should call `trackFinalResponseError` in the other places and remove the defer

Reply:
I applied the change. With the new change, the only difference is that previously if there were subgraph errors they were guaranteed to be the final error set to ctx.error, since it is not deferred anymore, this is not guaranteed.

Also for WriteError we already called `trackFinalResponseError` before WriteError in the existing code.